### PR TITLE
Allow to set the reply policy in a smart manner

### PR DESCRIPTION
**Pull request by MaximeMaW**
Depending on whether the original email was encrypted, one can set
different reply policies.

This fixes an issue raised in issue #865, when `mu4e-compose-crypto-reply-policy` conflicted with sign-by-default. Now, two variables replace the old `mu4e-compose-crypto-reply-policy`:

- mu4e-compose-crypto-reply-encrypted-policy
- mu4e-compose-crypto-reply-plain-policy

I'd be very happy to make any adjustment.

### CODE REVIEW

File: mu4e/mu4e-compose.el [hunk 1]
```diff
@@ -146,7 +146,7 @@ Also see `mu4e-context-policy'."
 	   :group 'mu4e-compose))
 
 
-(defcustom mu4e-compose-crypto-reply-policy 'sign-and-encrypt
+(defcustom mu4e-compose-crypto-reply-encrypted-policy 'sign-and-encrypt
   "Policy for signing/encrypting replies to encrypted messages.
 We have the following choices:
```
Comment (djcb) [inline mu4e/mu4e-compose.el]: I think we'd need a `make-obsolete-variable`, to warn users of the old variable.

File: mu4e/mu4e-compose.el [hunk 2]
```diff
@@ -162,6 +162,22 @@ We have the following choices:
 	   :safe 'symbolp
 	   :group 'mu4e-compose))
 
+(defcustom mu4e-compose-crypto-reply-plain-encrypted-policy 'sign  "Policy for signing/encrypting replies to encrypted messages.
+We have the following choices:
```
Comment (djcb) [inline mu4e/mu4e-compose.el]: this line is a bit too long - please limit it to 80 chars

File: mu4e/mu4e-compose.el [hunk 3]
```diff
@@ -525,13 +541,23 @@ buffers; lets remap its faces so it uses the ones for mu4e."
 
 (defun mu4e~compose-crypto-reply (parent compose-type)
   "When composing a reply to an encrypted message, we can
-automatically encrypt that reply."
-  (when (and  (eq compose-type 'reply)
+automatically encrypt that reply. When the message is unencrypted, 
+we can decide what we want to do."
+  (message "%S %S" parent compose-type)
+  (if (and  (eq compose-type 'reply)
```
Comment (djcb) [inline mu4e/mu4e-compose.el]: Those debug `message` things should probably go.

**Comment by djcb** [comment]
Anyway, apart from those details, looks good!

File: mu4e/mu4e-compose.el (hunk 1 above)
Comment (MaximeMaW) [reply mu4e/mu4e-compose.el]: I'm not sure about the standard way of doing that. I proposed something in commit  e761f74.

**Comment by MaximeMaW** [comment]
@djcb, any comment on the updated commits (e761f74)?

Thank you very much!

**Comment by djcb** [comment]
Merged, thanks!

**Pull request merged**